File: examples/legacy-architecture-samples/simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/injection/SchedulerModule.java
package com.zhuinden.simplestackexamplemvvm.application.injection;

import com.zhuinden.simplestackexamplemvvm.core.scheduler.BackgroundScheduler;
import com.zhuinden.simplestackexamplemvvm.core.scheduler.MainThreadScheduler;
import com.zhuinden.simplestackexamplemvvm.core.scheduler.NetworkScheduler;
import com.zhuinden.simplestackexamplemvvm.core.scheduler.Scheduler;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Owner on 2017. 07. 27..
 */

@Module
public class SchedulerModule {
    @Provides
    @Named("NETWORK")
    Scheduler networkScheduler(NetworkScheduler networkScheduler) {
        return networkScheduler;
    }

    @Provides
    @Named("BACKGROUND")
    Scheduler backgroundScheduler(BackgroundScheduler backgroundScheduler) {
        return backgroundScheduler;
    }

    @Provides
    @Named("MAIN_THREAD")
    Scheduler mainThreadScheduler(MainThreadScheduler mainThreadScheduler) {
        return mainThreadScheduler;
    }
}

File: samples/advanced-samples/mvvm-sample/src/main/java/com/zhuinden/simplestackexamplemvvm/application/injection/SchedulerModule.kt
package com.zhuinden.simplestackexamplemvvm.application.injection


import com.zhuinden.simplestackexamplemvvm.core.scheduler.BackgroundScheduler
import com.zhuinden.simplestackexamplemvvm.core.scheduler.MainThreadScheduler
import com.zhuinden.simplestackexamplemvvm.core.scheduler.NetworkScheduler
import com.zhuinden.simplestackexamplemvvm.core.scheduler.Scheduler
import dagger.Module
import dagger.Provides
import javax.inject.Named

/**
 * Created by Owner on 2017. 07. 27..
 */
@Module
class SchedulerModule {
    @Provides
    @Named("NETWORK")
    fun networkScheduler(networkScheduler: NetworkScheduler): Scheduler = networkScheduler

    @Provides
    @Named("BACKGROUND")
    fun backgroundScheduler(backgroundScheduler: BackgroundScheduler): Scheduler = backgroundScheduler

    @Provides
    @Named("MAIN_THREAD")
    fun mainThreadScheduler(mainThreadScheduler: MainThreadScheduler): Scheduler = mainThreadScheduler
}